src/python/pants/backend/jvm/subsystems/zinc.py

```diff
@@ -92,9 +110,29 @@ def register_options(cls, register):
       cls.register_jvm_tool(register,
                             Zinc.ZINC_EXTRACTOR_TOOL_NAME,
                             classpath=[
-                              JarDependency('org.pantsbuild', 'zinc-extractor_2.11', '0.0.4')
+                              JarDependency('org.pantsbuild', 'zinc-extractor_2.11', '0.0.6')
                             ])
 
+      # Register tools for fixed versions of Scala, 2.10, 2.11 and 2.12.
+      # Relies on ScalaPlatform to get the revision version from the major.minor version.
+      # The tool with the correct scala version will be retrieved later,
+      # taking the user-passed option into account.
+      def register_scala_tools_for_versions(tools, scala_versions):
+        for tool_name in tools:
+          for scala_version in scala_versions:
+            cls.register_jvm_tool(register,
+                                  ScalaPlatform.versioned_tool_name(tool_name, scala_version),
+                                  classpath=[
+                                    ScalaPlatform
+                                      .create_jardep(tool_name, scala_version)
+                                      .copy(intransitive=True)
+                                  ])
+
+      register_scala_tools_for_versions(
```

Comment:
Since you're calling this method with lists, it doesn't look like it needs to be a method. Alternatively, could move the loops out and have just the `register` call be wrapped in a method.